servicetalk-grpc-netty/src/test/java/io/servicetalk/grpc/netty/ProtocolCompatibilityTest.java

```diff
@@ -569,53 +589,131 @@ void serviceTalkToServiceTalkErrorWithStatusViaServerFilter(
-        testGrpcError(client, server, false, streaming, compression, GrpcStatusCode.DEADLINE_EXCEEDED, null);
+        try (ServerContext proxyCtx = buildTimeoutProxy(server.listenAddress(), ssl)) {
+            final CompatClient client = serviceTalkClient(proxyCtx.listenAddress(), ssl, compression, DEFAULT_DEADLINE);
+            testGrpcError(client, server, false, streaming, compression, DEADLINE_EXCEEDED, null);
+        }
     }
 
     @ParameterizedTest
     @MethodSource("sslStreamingAndCompressionParams")
-    @Disabled("https://github.com/apple/servicetalk/issues/1489")
-    void grpcJavaToServiceTalkTimeout(final boolean ssl,
-                                      final boolean streaming,
-                                      final String compression) throws Exception {
-        Duration timeout = Duration.ofNanos(1);
+    void grpcJavaToServiceTalkTimeout(final boolean ssl, final boolean streaming, final String compression)
+            throws Exception {
         final TestServerContext server = serviceTalkServer(ErrorMode.NONE, ssl, compression, null);
-        final CompatClient client = grpcJavaClient(server.listenAddress(), compression, ssl, timeout);
-        testGrpcError(client, server, false, streaming, compression, GrpcStatusCode.DEADLINE_EXCEEDED, null);
+        try (ServerContext proxyCtx = buildTimeoutProxy(server.listenAddress(), ssl)) {
+            final CompatClient client = grpcJavaClient(proxyCtx.listenAddress(), compression, ssl, DEFAULT_DEADLINE);
+            testGrpcError(client, server, false, streaming, compression, DEADLINE_EXCEEDED, null);
+        }
     }
 
     @ParameterizedTest
     @MethodSource("sslStreamingAndCompressionParams")
-    @Disabled("https://github.com/apple/servicetalk/issues/1489")
-    void serviceTalkToServiceTalkTimeout(final boolean ssl,
-                                         final boolean streaming,
-                                         final String compression) throws Exception {
-        Duration timeout = Duration.ofNanos(1);
+    void serviceTalkToServiceTalkTimeout(final boolean ssl, final boolean streaming, final String compression)
+            throws Exception {
         final TestServerContext server = serviceTalkServer(ErrorMode.NONE, ssl, compression, null);
-        final CompatClient client = serviceTalkClient(server.listenAddress(), ssl, compression, timeout);
-        testGrpcError(client, server, false, streaming, compression, GrpcStatusCode.DEADLINE_EXCEEDED, null);
+        try (ServerContext proxyCtx = buildTimeoutProxy(server.listenAddress(), ssl)) {
+            final CompatClient client = serviceTalkClient(proxyCtx.listenAddress(), ssl, compression, DEFAULT_DEADLINE);
+            testGrpcError(client, server, false, streaming, compression, DEADLINE_EXCEEDED, null);
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("sslAndStreamingParams")
```

Comment:
While `sslAndStreamingParams` produces 4 pairs of booleans, the method name of the source does not match with what params the test defines. Consider using:
```java
@CsvSource({"false,false", "false,true", "true,false", "true,true"})
```